internal/resource/route.go

```diff
@@ -102,17 +116,58 @@ func (r Route) ProcessDefaults(ctx context.Context) error {
 		return fmt.Errorf("invalid nil resource")
 	}
 	if len(r.Route.Protocols) == 0 {
-		err := mergo.Merge(r.Route, defaultRoute,
-			mergo.WithTransformers(wrappersPBTransformer{}))
-		if err != nil {
-			return err
+		if r.Route.Expression == "" {
+			err := mergo.Merge(r.Route, defaultRoute,
+				mergo.WithTransformers(wrappersPBTransformer{}))
+			if err != nil {
+				return err
+			}
+		} else {
+			err := mergo.Merge(r.Route, defaultExpressionRoute,
+				mergo.WithTransformers(wrappersPBTransformer{}))
+			if err != nil {
+				return err
+			}
 		}
 	}
 	defaultID(&r.Route.Id)
 	return nil
 }
 
+func buildSchema() *goatcrouter.Schema {
+	schema := goatcrouter.NewSchema()
+
+	for _, fieldname := range []string{
```

Comment:
nit: might be better to use `map[goatcrouter.FieldType][]string`

Reply:
agreed.  that was a direct translation of the current Lua code, and should be kept in sync with that.  shouldn't be difficult as long as the list of predefined fields is visible in any syntax highlighter.